state/mock.go

```diff
@@ -26,14 +27,18 @@ type MockState struct {
 	InvalidBlockHash     crypto.Hash
 	Committee            *committee.Committee
 	Lock                 deadlock.RWMutex
+	AddToPool            bool
+	BroadCastTx          bool
 }
 
 func MockingState(committee *committee.Committee) *MockState {
 	return &MockState{
-		GenHash:   crypto.GenerateTestHash(),
-		Store:     store.MockingStore(),
-		TxPool:    txpool.MockingTxPool(),
-		Committee: committee,
+		GenHash:     crypto.GenerateTestHash(),
+		Store:       store.MockingStore(),
+		TxPool:      txpool.MockingTxPool(),
+		Committee:   committee,
+		AddToPool:   true,
+		BroadCastTx: true,
```

Comment:
About `AddToPool` and `BroadCastTx`:

I have a suggestion here. If a transaction already exists inside pool, Add and Broadcast should return an error.
By this check we can omit these two variables which look unnecessary to me.